src/screens/DomainScreen/index.js

```diff
@@ -101,20 +101,21 @@ const DomainScreen = () => {
   };
 
   const getIFollow = async () => {
-    if (ifollow.length === 0) {
+    const isFollow = JSON.stringify(ifollow.map((i) => i.domain_id_followed)).includes(iddomain);
+    if (!isFollow) {
       const res = await getDomainIdIFollow();
       setIFollow(res.data, dispatch);
     } else {
-      setFollow(JSON.stringify(ifollow).includes(iddomain));
+      setFollow(isFollow);
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for checking if a domain is followed has been changed. Previously, it was checked if the `ifollow` array is empty or not. Now, it checks if the `iddomain` exists in the `ifollow` array. This change seems to be more accurate as it directly checks for the existence of the specific domain in the `ifollow` array. However, using `JSON.stringify()` and `includes()` to check for the existence of an item in an array is inefficient and can lead to false positives if there are similar substrings in the array. A better approach would be to use the `some()` method which stops at the first match.

```diff
- const isFollow = JSON.stringify(ifollow.map((i) => i.domain_id_followed)).includes(iddomain);
+ const isFollow = ifollow.some((i) => i.domain_id_followed === iddomain);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->